test/subtype.jl

```diff
@@ -796,6 +800,11 @@ function test_intersection()
                    (@UnionAll X<:(@UnionAll T Tuple{T,T}) Ref{X}))
     @testintersect(Ref{@UnionAll T @UnionAll S Tuple{T,S}},
                    Ref{@UnionAll T Tuple{T,T}}, Bottom)
+
+    @testintersect(Tuple{T,T} where T<:Union{UpperTriangular, UnitUpperTriangular},
+                   Tuple{AbstractArray{T,N}, AbstractArray{T,N}} where N where T,
```

Comment:
Yes, I'm pretty sure that syntax is possible. Mostly we need to decide the nesting order. With nested `where`, it's clear that the last one is the outermost. However in type declarations you can write `type Foo{T<:Real, S<:Vector{T}}`; it seems more natural for the first variable to be the outermost. The question is which one `where T, S` corresponds to.